pandas/core/arrays/boolean.py

```diff
@@ -382,9 +384,14 @@ def to_numpy(
         if dtype is None:
             dtype = object
         if self._hasna:
-            if is_bool_dtype(dtype) and na_value is libmissing.NA:
+            if (
+                not (is_object_dtype(dtype) or is_string_dtype(dtype))
```

Comment:
It's a bit easiser / nicer to essentially whitelist the allowed types NA can convert to (object, str), rather than just checking bool and letting NumPy raise for invalid types.

We get to share the implementation with IntegerArray, and we get more consistent error messages.